src/Bridge/Symfony/Bundle/DataCollector/RequestDataCollector.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DataCollector;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+final class RequestDataCollector extends DataCollector
+{
+    private $collectionFactory;
+
+    private $metadataFactory;
+
+    public function __construct(ResourceNameCollectionFactoryInterface $collectionFactory, ResourceMetadataFactoryInterface $metadataFactory)
+    {
+        $this->collectionFactory = $collectionFactory;
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    public function collect(Request $request, Response $response, \Exception $exception = null): void
+    {
+        $resourceClass = $request->attributes->get('_api_resource_class');
+        $resourceMetadata = $resourceClass ? $this->metadataFactory->create($resourceClass) : null;
+
+        $this->data = [
+            'resource_class' => $resourceClass,
+            'resource_metadata' => $resourceMetadata,
+            'method' => $request->getMethod(),
+            'acceptable_content_types' => $request->getAcceptableContentTypes(),
+        ];
+    }
+
+    public function getMethod(): string
+    {
+        return $this->data['method'];
```

Comment:
They always exists (if `collect` has been called), as [Request::getMethod](http://api.symfony.com/3.4/Symfony/Component/HttpFoundation/Request.html#method_getMethod) returns a string

Reply:
> if `collect` has been called

That is what I wanted to point out! Perhaps an exception would be more appropriate than a fatal error in this particular case.

Reply:
WDYT?

Reply:
Oops, I missed that one today :), will be OK in the next return fixes in a few minutes :)

Reply:
@meyerbaptiste done, But I preferred having a default value to `''` because it's seems really overkill to throw an exception for something that should never happened.

Yes, theoretically, `collect()` may not have been called, but in that case, the developer probably made something really unusual. If Symfony manage the datacollectors, then everything is OK.